src/widgets/piechart.rs

```diff
@@ -0,0 +1,109 @@
+use std::cmp::min;
+use std::f64::consts::PI;
+
+use crate::buffer::Buffer;
+use crate::layout::Rect;
+use crate::style::{Color, Style};
+use crate::symbols::bar;
+use crate::widgets::{Block, Widget};
+
+const TAU: f64 = PI * 2.0;
+
+#[derive(Debug, Clone)]
+pub struct PieChart<'a> {
+    block: Option<Block<'a>>,
+    angles: Vec<((f64, f64), Color)>,
+}
+
+impl<'a> Default for PieChart<'a> {
+    fn default() -> PieChart<'a> {
+        PieChart {
+            block: None,
+            angles: Vec::new(),
+        }
+    }
+}
+
+impl<'a> PieChart<'a> {
+    pub fn data(mut self, data: &'a [(f64, Color)]) -> PieChart<'a> {
```

Comment:
We may want to have some sort of labels or legend. Otherwise, how are we going to know which value is what ?

Reply:
My concern with adding a legend is that it will take up space inside the block. Perhaps a separate legend `Widget` could be added to let the user decide how much terminal space should be allocated?

Reply:
I agree that a legend might be tricky to implement so we can leave it out of this PR but I still think that we should at least have a way to show some labels (e.g percentages) on the pie chart itself, no ?

Reply:
I was worried that labels on the chart itself would take up too much space inside the chart. Once I fix the horizontal spacing, I'll test it out, but I'm worried it could result in smaller circles having too many "pixels" taken up by labels to be readable.

Reply:
I could probably make it optional whether percentages/labels are rendered or not.